src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -702,6 +701,15 @@ public static ReadOnlyMemory<char> AsMemory(this string? text, Range range)
                     return IndexOfAnyExcept(span, values[0], values[1], values[2], values[3]);
 
                 default:
+                    if (RuntimeHelpers.IsBitwiseEquatable<T>() && Unsafe.SizeOf<T>() == sizeof(short))
```

Comment:
Nit: since we're unsafe casting to char, should we compare against sizeof(char), just for consistency?  Same below.